### PR TITLE
chore: document subgraph schema

### PR DESCRIPTION
**Pull request by banasa44**
## Description

2nd part of the inline comments for subgraph schema. These are meant to be reflected in our auto-generated documentation (subgraph/reference-guide) in our developer-portal.

https://subgraph.satsuma-prod.com/aragon/osx-carles-sepolia/playground

Task ID: [OS-1206](https://aragonassociation.atlassian.net/browse/OS-1206)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.
- [x] I have updated the Subgraph and added a QA URL to the description of this PR.

[OS-1206]: https://aragonassociation.atlassian.net/browse/OS-1206?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/schema.graphql [hunk 1]
```diff
@@ -462,152 +462,356 @@ type StandardCallback @entity {
   magicNumber: Bytes!
 }
 
-# DAO
+#############################
+########### DAO #############
+#############################
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:467]: This pattern is only for us to see the sections better ? or it will contribute to the dev-portal in some way?

File: packages/subgraph/schema.graphql [hunk 2]
```diff
@@ -462,152 +462,356 @@ type StandardCallback @entity {
...
 
+"""
+The Aragon DAOs deployed using the `DAOFactory` SC `createDAO` function and registered in the `DAORegistry` SC. From this entity you are able to query
+any relevant information related to your DAO.
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:471]:
```suggestion
any relevant information related to any DAO.
```

File: packages/subgraph/schema.graphql [hunk 3]
```diff
@@ -462,152 +462,356 @@ type StandardCallback @entity {
...
+"""
 type Dao @entity {
-  id: ID! # use address as id
+  " The address of the deployed DAO contract registered in the `DAORegistry` SC. "
+  id: ID!
+
+  " ENS DAO subdomain, if added while creating the DAO. "
   subdomain: String
+
+  " The address of the creator who called (msg.sender) `createDao` in the `DAOFactory` SC. "
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:480]:
```suggestion
  " The address of the creator (msg.sender) who called `createDao` in the `DAOFactory` SC. "
```

File: packages/subgraph/schema.graphql [hunk 4]
```diff
@@ -462,152 +462,356 @@ type StandardCallback @entity {
...
   creator: Bytes!
+
+  " The DAO uri used with [EIP-4824](https://eips.ethereum.org/EIPS/eip-4824), if added while creating the DAO. "
   daoURI: String
+
+  " Additional information related to the created DAO, if added while creating the DAO. "
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:486]: perhaps instead of `if added` we can use the wording of `optional`?

File: packages/subgraph/schema.graphql [hunk 5]
```diff
@@ -462,152 +462,356 @@ type StandardCallback @entity {
...
   metadata: String
+
+  " Timestamp for the block where the DAO was created. "
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:489]:
```suggestion
  " Timestamp of the block where the DAO was created. "
```

File: packages/subgraph/schema.graphql [hunk 6]
```diff
@@ -462,152 +462,356 @@ type StandardCallback @entity {
...
   createdAt: BigInt!
+
+  " Tokens held by the DAO. "
   token: ERC20Contract
+
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:494]: @Michael-A-Heuer for visibility
this attribute is not accurate any more, but also need some planning

File: packages/subgraph/schema.graphql [hunk 7]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " Actions executed by the DAO `execute` function (refer to `Objects/Action` in this documentation site for more information). "
   actions: [Action!]! @derivedFrom(field: "dao")
+
+  " Token transfers associated with the DAO (refer to `Interfaces/TokenTransfer` in this documentation site for more information). "
   transfers: [TokenTransfer!]! @derivedFrom(field: "dao")
+
+  " DAO balances for all kinds of tokens (refer to `Interfaces/TokenBalance` in this documentation site for more information). "
   balances: [TokenBalance!]! @derivedFrom(field: "dao")
+
+  " The permissions given to a DAO to govern its actions (refer to `Objects/Permission` in this documentation site for more information). "
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:494]: only commenting it here
thinks like : `(refer to `Objects/Permission` in this documentation site for more information)` does this create an actual link?

File: packages/subgraph/schema.graphql [hunk 8]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
   permissions: [Permission!]! @derivedFrom(field: "dao")
+
+  " The proposals to execute actions that a DAO has had. "
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:497]: needs a better writing IMO, perhaps something like `A list of proposals for actions to be executed by the DAO, including both open and executed proposals.`

File: packages/subgraph/schema.graphql [hunk 9]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
   proposals: [IProposal!]! @derivedFrom(field: "dao")
+
+  " The address of the trusted forwarder required for meta transactions, if added while creating the DAO. "
   trustedForwarder: Bytes
+
+  " Deprecated field. "
   signatureValidator: Bytes
+
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:505]: @Michael-A-Heuer what should we do with `trustedForwarder` ? is't this also to be depricated

File: packages/subgraph/schema.graphql [hunk 10]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
   magicNumber: Bytes!
 }
 
-# DAO
+#############################
+########### DAO #############
+#############################
 
+"""
+The Aragon DAOs deployed using the `DAOFactory` SC `createDAO` function and registered in the `DAORegistry` SC. From this entity you are able to query
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:460]: SC refers to Smart Contract presumably?

File: packages/subgraph/schema.graphql [hunk 11]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " The address of the trusted forwarder required for meta transactions, if added while creating the DAO. "
   trustedForwarder: Bytes
+
+  " Deprecated field. "
   signatureValidator: Bytes
+
+  " Registered callback functions supported by the DAO (refer to `Objects/StandardCallback` in this documentation site for more information). "
   standardCallbacks: [StandardCallback!]! @derivedFrom(field: "dao")
 
+  " The plugins that had been installed/uninstalled to the DAO. "
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:509]: Not sure if the include preparations as well, can u verify

File: packages/subgraph/schema.graphql (hunk 10 above)
Comment (jordaniza) [reply packages/subgraph/schema.graphql:460]: If so, I'd remove it

File: packages/subgraph/schema.graphql (hunk 6 above)
Comment (jordaniza) [reply packages/subgraph/schema.graphql:484]: Why not?

File: packages/subgraph/schema.graphql [hunk 12]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " Tokens held by the DAO. "
   token: ERC20Contract
+
+  " Actions executed by the DAO `execute` function (refer to `Objects/Action` in this documentation site for more information). "
   actions: [Action!]! @derivedFrom(field: "dao")
+
+  " Token transfers associated with the DAO (refer to `Interfaces/TokenTransfer` in this documentation site for more information). "
   transfers: [TokenTransfer!]! @derivedFrom(field: "dao")
+
+  " DAO balances for all kinds of tokens (refer to `Interfaces/TokenBalance` in this documentation site for more information). "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:491]:
```suggestion
  " Balance for tokens transferred to the DAO using tracked methods (refer to `Interfaces/TokenBalance` in this documentation site for more information). "
```

File: packages/subgraph/schema.graphql (hunk 1 above)
Comment (banasa44) [reply packages/subgraph/schema.graphql:457]: just for us

File: packages/subgraph/schema.graphql [hunk 13]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " Timestamp for the block where the DAO was created. "
   createdAt: BigInt!
+
+  " Tokens held by the DAO. "
   token: ERC20Contract
+
+  " Actions executed by the DAO `execute` function (refer to `Objects/Action` in this documentation site for more information). "
   actions: [Action!]! @derivedFrom(field: "dao")
+
+  " Token transfers associated with the DAO (refer to `Interfaces/TokenTransfer` in this documentation site for more information). "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:488]:
```suggestion
  " Token transfers associated with the DAO using tracked methods (refer to `Interfaces/TokenTransfer` in this documentation site for more information). "
```

File: packages/subgraph/schema.graphql [hunk 14]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " DAO balances for all kinds of tokens (refer to `Interfaces/TokenBalance` in this documentation site for more information). "
   balances: [TokenBalance!]! @derivedFrom(field: "dao")
+
+  " The permissions given to a DAO to govern its actions (refer to `Objects/Permission` in this documentation site for more information). "
   permissions: [Permission!]! @derivedFrom(field: "dao")
+
+  " The proposals to execute actions that a DAO has had. "
   proposals: [IProposal!]! @derivedFrom(field: "dao")
+
+  " The address of the trusted forwarder required for meta transactions, if added while creating the DAO. "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:500]:
```suggestion
  " The address of the trusted forwarder required for meta transactions, if added when creating the DAO. "
```

File: packages/subgraph/schema.graphql (hunk 11 above)
Comment (jordaniza) [inline packages/subgraph/schema.graphql:509]:
```suggestion
  " The plugins that have been installed/uninstalled on the DAO. "
```

File: packages/subgraph/schema.graphql [hunk 15]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+
+  " Registered callback functions supported by the DAO (refer to `Objects/StandardCallback` in this documentation site for more information). "
   standardCallbacks: [StandardCallback!]! @derivedFrom(field: "dao")
 
+  " The plugins that had been installed/uninstalled to the DAO. "
   plugins: [PluginInstallation!]! @derivedFrom(field: "dao")
 }
 
-# ACL
+#ACL
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:513]: PermissionManager?

File: packages/subgraph/schema.graphql [hunk 16]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " The address of the target contract to grant `who` permission to. "
   where: Bytes!
+
+  " Bytes32, the Keccak256 hash function of the ID for the permission. Normally the Keccak256 of the permission name (string). "
   permissionId: Bytes!
+
+  " The address (EOA or contract) to which the permission has been granted. "
   who: Bytes!
+
+  " The address of the context in which the permission has been granted. "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:537]: Can you write that in a slightly clearer way please?

aka: please provide more context on what you mean by 'context'

File: packages/subgraph/schema.graphql [hunk 17]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
   actor: Bytes!
+
+  " The address of the contract containing the conditions which authorize a call by another address. This contract must inherit from `PermissionCondition` and implement the `IPermissionCondition` interface. "
   condition: Bytes
 
-  # The following attributes is used to create
+  # The following attributes are used to create
   # one-to-many between DAO or PluginRepo to Permission
+
+  " The DAO that grants the permission (DAO uses `PermissionManager` SC `grant` function). "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:546]: I'd avoid `SC` as per above - it's implied when you're already reading about a subgraph

File: packages/subgraph/schema.graphql [hunk 18]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  id: ID!
+
+  " Each PluginRepo has to have a valid ENS subdomain associated, added while creating and registering it. "
   subdomain: String!
+
+  " The different incompatible versions for the associated plugin.  "
   releases: [PluginRelease!]! @derivedFrom(field: "pluginRepo")
-  # Holds all preparations. Also applied ones.
+
+  " The prepared necessary settings and addresses to install/uninstall to a DAO the plugin associated to a `PluginRepo`. Also applied ones. "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:570]:
```suggestion
  " Preparation settings and addresses required to install/uninstall a plugin from this repo on a DAO. These include settings which have previously been applied.  "
```

File: packages/subgraph/schema.graphql [hunk 19]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
   preparations: [PluginPreparation!]! @derivedFrom(field: "pluginRepo")
-  # Holds all installed and uninstalled installations.
+
+  # The plugins installed and uninstalled to DAOs associated to each `PluginRepo`.
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:573]:
```suggestion
  # List of installations and uninstallations for a given `PluginRepo`. 
```

File: packages/subgraph/schema.graphql [hunk 20]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
   installations: [PluginInstallation!] @derivedFrom(field: "appliedPluginRepo")
 
+  " The permissions given to a PluginRepo to govern its actions (e.g. MAINTAINER_PERMISSION_ID to create a new plugin version). "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:576]:
```suggestion
  " The permissions given to a `PluginRepo` to govern its actions (e.g. `MAINTAINER_PERMISSION_ID` to create a new plugin version). "
```

File: packages/subgraph/schema.graphql [hunk 21]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
   permissions: [Permission!]! @derivedFrom(field: "pluginRepo")
 }
 
+"""
+The contracts that will deploy your plugin using different factory patterns (new/minimal proxy pattern/UUPS pattern).
+Pointing to a specific version of the plugin (the address for one specific implementation contract).
+"""
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:583]: Try and avoid the implementation details if you can. The above I feel could be simplified.

File: packages/subgraph/schema.graphql [hunk 22]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
-  id: ID! # contract address
+  " The PluginSetup contract address. "
+  id: ID!
+
+  " The version of the plugin internally referenced in the PluginSetup. "
   versions: [PluginVersion!]! @derivedFrom(field: "pluginSetup")
 }
 
+"""
+Registration of the different incompatible versions of a plugin.
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:593]: Maybe say here that a release is dictated by a new version of a plugin being incompatible with an old version. Maybe also link to the Sem ver or releases section in the docs.

File: packages/subgraph/schema.graphql [hunk 23]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  metadata: String!
+
+  " Record for different compatible versions for the plugin. "
   builds: [PluginVersion!]! @derivedFrom(field: "release")
 }
 
+"""
+All the versions of a plugin, managed by the [`PluginRepo` SC](https://devs.aragon.org/docs/osx/how-it-works/framework/plugin-management/plugin-repo/).
+"""
 type PluginVersion @entity(immutable: true) {
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:615]: yeah this is a little better IMO

File: packages/subgraph/schema.graphql [hunk 24]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+"""
+The previous step to installing/uninstalling a plugin to/from your DAO. All `PluginSetup` SCs must implement the prepare functions containing
+necessary permissions and settings to install/uninstall a plugin.
+"""
 type PluginPreparation @entity(immutable: true) {
-  id: ID! # psp setupId + installationId
+  " Concatenation of DAO address, plugin address and the preparedSetupId. Joined by '_'. "
+  id: ID!
+
+  " The application of this preparation step. "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:690]: Can you rephrase?

File: packages/subgraph/schema.graphql [hunk 25]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
+  " The bytes-encoded data containing the input parameters for the installation as specified in the plugin's build metadata JSON file. "
   data: Bytes
+
+  " The deployed plugin SC by the `PluginSetup` SC. "
   pluginAddress: Bytes!
+
+  " The address array of helpers (contracts or EOAs) associated with this plugin version. "
   helpers: [Bytes!]!
+
+  " The array of multi-targeted permission operations to be applied by the `PluginSetupProcessor` to install/uninstall/update the plugin in the DAO. "
```
Comment (jordaniza) [inline packages/subgraph/schema.graphql:717]: I don't think you strictly need to apply multi target permissions

File: packages/subgraph/schema.graphql (hunk 15 above)
Comment (Rekard0) [inline packages/subgraph/schema.graphql:513]:
```suggestion
# Permissions
```
shouldn't we do a similar pattern as you proposed for the DAO section?

File: packages/subgraph/schema.graphql [hunk 26]
```diff
@@ -462,152 +452,356 @@ type StandardCallback @entity {
...
 # Cannot be immutable because a permission can be revoked and granted in the same TX.
 # This results in an error in the subgraph if the entity is immutable.
 # No need to store granted as we can delete permission when revoked
+
+"""
+[Aragon permission system](https://devs.aragon.org/docs/osx/how-it-works/core/permissions/), permissions between contracts and wallets allow
+our DAOs (and `PluginRepo`s) to manage and govern its actions. A permission is the hash of a struct which specifies an address `who` being
+allowed to call certain functions on a contract address `where`.
+"""
 type Permission @entity {
```
Comment (Rekard0) [inline packages/subgraph/schema.graphql:524]: Are we sure this statement is true? I don't think so

File: packages/subgraph/schema.graphql (hunk 4 above)
Comment (banasa44) [reply packages/subgraph/schema.graphql:476]: Rephrased like this:
.... , -> .... . Optional to add when creating the DAO.

File: packages/subgraph/schema.graphql (hunk 25 above)
Comment (banasa44) [reply packages/subgraph/schema.graphql:717]: We are inside PluginPreparation entity, which has permissions that will be handled by the `psp` which only uses `applyMultiTargetPermission` function
